src/ruby/yast/rspec/helpers.rb

```diff
@@ -9,60 +9,76 @@ module Helpers
       # When the requested module is present in the system then it is imported
       # as usually. If it is missing a substitute definition is created.
       #
+      # In the substitute definition it is enough to define only the methods
+      # used in the tests, you do not need to mirror the complete API.
+      #
       # @note This needs to be called *after* starting code coverage with
       #   `SimpleCov.start` so the coverage of the imported modules is correctly
       #   counted.
       #
       # @note You can force using the defined stubs although the modules are
       #   present in the system, this might be useful to actually test the stubs.
-      #
-      # @example Mock the `Language` module and the `Language.language` method
-      # Yast::RSpec::Helpers.define_yast_module("Language") do
-      #   # see modules/Language.rb in yast2-country
-      #   module Yast
-      #     class LanguageClass < Module
-      #       # @return [String]
-      #       def language; end
-      #     end
-      #     Language = LanguageClass.new
-      #   end
-      # end
+      #   Just the define the YAST_FORCE_MODULE_STUBS environment variable.
       #
       # @example Mock empty `AutoInstall` module
       # Yast::RSpec::Helpers.define_yast_module("AutoInstall")
       #
+      # @example Mock the `Language` module with the `language` method
+      # Yast::RSpec::Helpers.define_yast_module("Language", methods: [:language])
+      #
+      # @example Mock the `AutoinstStorage` module with `Import` taking one parameter
+      # Yast::RSpec::Helpers.define_yast_module("AutoinstStorage") do
+      #   def Import(_config); end
+      # end
+      #
       # @param name [String] name of the YaST module
-      # @param block [Block] optional module definition, it should provide
-      #   the same API as the original module, it is enough to define only the
-      #   methods used in the tests, if no block is passed an empty module is defined
-      def self.define_yast_module(name, &block)
+      # @param methods [Array<Symbol>] optional list of defined methods,
+      #   the defined methods accept no parameter, if you need a parameter
```

Comment:
maybe this can be easily fixed by 
```
          methods.each { |m| define_singleton_method(m) {|*_a, **_k, &_b| } }
```
this way it accepts everything

Reply:
IIRC the RSpec verifying doubles not only check that the mocked method exists but also checks the arity of the method. So that would very likely break with this change.